src/style.css

```diff
@@ -0,0 +1,38 @@
+* {
+  box-sizing: border-box;
+  margin: 0;
+}
+
+.stars {
+  display: flex;
+}
+
+.stars .stars__star {
+  border-radius: 0.5px;
+  width: 16px;
+  height: 16px;
+  background-image: url(./images/star-active.svg);
+  background-repeat: no-repeat;
+  background-position: center;
+}
+
+.stars--0 .stars__star {
+  border-radius: 0.5px;
+  width: 16px;
+  height: 16px;
+  background-image: url(./images/star.svg);
+  background-repeat: no-repeat;
+  background-position: center;
+}
+
+.stars--1 .stars__star:not(:first-child),
+.stars--2 .stars__star:not(:nth-of-type(-n+2)),
+.stars--3 .stars__star:not(:nth-of-type(-n+3)),
+.stars--4 .stars__star:not(:nth-of-type(-n+4)),
+.stars--5 .stars__star:not(:nth-of-type(-n+5)) {
+  background-image: url(./images/star.svg);
+}
+
+.stars .stars__star:not(:last-child) {
```

Comment:
You don't need to use additional class `.stars` here. 
Try just to nest this rule into `.stars__star` selector

```
.stars__star {
  &:not(:last-child) {
    ...
  }
}
```